src/lm_buddy/jobs/_entrypoints/prometheus.py

```diff
@@ -0,0 +1,165 @@
+
+    return feedback, score
+
+
+def instruction_to_prompt(instruction: str) -> str:
+    """Given some text containing Prometheus instructions, transform it
+    into an actual llama-2 prompt.
+    """
+    conv = get_conv_template("llama-2")
+    conv.set_system_message("You are a fair evaluator language model.")
+    conv.append_message(conv.roles[0], instruction)
+    conv.append_message(conv.roles[1], None)
+    return conv.get_prompt()
+
+
+def run_eval(
+    config: PrometheusJobConfig, data: Dataset, tokenizer: PreTrainedTokenizer, client: OpenAI
+) -> str:
+    # enable / disable tqdm
+    dataset_iterable = tqdm(data) if config.evaluation.enable_tqdm else data
+
+    # open the output file for writing and iterate on samples
+    tracking_name = config.tracking.name if config.tracking is not None else "output.json"
+    output_fname = Path(config.evaluation.tmp_folder) / tracking_name
+    with output_fname.open("w") as file:
+        for sample in dataset_iterable:
+            # convert instructions from the dataset (`text_field` in a dict) to
+            # prompts that prometheus accepts
+            prompt = instruction_to_prompt(sample[config.dataset.text_field])
+
+            # skip those examples which are too long
+            tokenized_prompt = tokenizer(prompt, truncation=False)
+            if len(tokenized_prompt["input_ids"]) > 3072:
+                continue
+
+            # prepare output
+            result = copy.deepcopy(sample)
+            result["prometheus_output"] = []
+            result["prometheus_score"] = []
+
+            for idx in range(config.evaluation.num_answers):
+                i = 0
+                while i < config.evaluation.max_retries:
+                    try:
+                        response = openai_completion(config, client, prompt)
+                        feedback, score = parse_response(config, response)
+                        break
+                    except (OpenAIError, BadResponseError) as e:
+                        print(f"[w] {e.message}, retrying ({i+1}/{config.evaluation.max_retries})")
+                        i += 1
+                        if i == config.evaluation.max_retries:
+                            raise e
+
+                result["prometheus_output"].append(feedback)
+                result["prometheus_score"].append(score)
+
+            # dump sample results incrementally
+            file.write(json.dumps(result) + "\n")
+
+    # convert plain json dataset in HF format
+    output_hf_name = str(Path(config.evaluation.tmp_folder) / "hf" / tracking_name)
+    ds = load_dataset("json", data_files=str(output_fname), split="train")
+    ds.save_to_disk(output_hf_name)
+
+    return str(output_hf_name)
+
+
+def run_prometheus(config: PrometheusJobConfig, artifact_loader: ArtifactLoader):
+    # load dataset from W&B artifact
+    hf_loader = HuggingFaceAssetLoader(artifact_loader)
+    data = hf_loader.load_dataset(config.dataset)
```

Comment:
This line should be removed within the scope of the initialized W&B run, or else artifact lineages will not be connected. The tracking run must be active to link an input artifact (the dataset in this case).

Reply:
Moving the dataset and tokenizer loading to within the `run_eval` function seems most appropriate, since that is only called after run initialization.

Reply:
Good catch, I fixed it and now have lineage working:
<img width="976" alt="image" src="https://github.com/mozilla-ai/lm-buddy/assets/4476645/ce31b7bd-6af4-415d-b7db-028e48b3da5a">